impeller/renderer/backend/vulkan/command_pool_vk.h

```diff
@@ -5,36 +5,80 @@
 
 namespace impeller {
 
-class ContextVK;
-
-class CommandPoolVK {
+//------------------------------------------------------------------------------
+/// @brief      An opaque object that provides |vk::CommandBuffer| objects.
+///
+/// A best practice is to create a |CommandPoolVK| for each thread that will
+/// submit commands to the GPU, and to recycle (reuse) |vk::CommandBuffer|s by
+/// resetting them in a background thread.
+///
+/// @see
+/// https://arm-software.github.io/vulkan_best_practice_for_mobile_developers/samples/performance/command_buffer_usage/command_buffer_usage_tutorial.html#resetting-the-command-pool
+class CommandPoolVK final {
  public:
+  /// @brief      Gets the |CommandPoolVK| for the current thread.
+  ///
+  /// @param[in]  context  The |ContextVK| to use.
+  ///
+  /// If the current thread does not have a |CommandPoolVK|, one will be created
+  /// and returned. If the current thread already has a |CommandPoolVK|, it will
+  /// be returned.
+  ///
+  /// @return     The |CommandPoolVK| for the current thread, or |nullptr| if
+  ///             either the |ContextVK| is invalid, or a pool could not be
+  ///             created for any reason.
+  ///
+  ///             In other words an invalid command pool will never be returned.
   static std::shared_ptr<CommandPoolVK> GetThreadLocal(
       const ContextVK* context);
 
+  /// @brief      Clears all |CommandPoolVK|s for the given |ContextVK|.
+  ///
+  /// @param[in]  context  The |ContextVK| to clear.
+  ///
+  /// Every |CommandPoolVK| that was created for every thread that has ever
+  /// called |GetThreadLocal| with the given |ContextVK| will be cleared.
+  ///
+  /// @note       Should only be called when the |ContextVK| is being destroyed.
   static void ClearAllPools(const ContextVK* context);
 
   ~CommandPoolVK();
 
+  /// @brief      Whether or not this |CommandPoolVK| is valid.
+  ///
+  /// A command pool is no longer when valid once it's been |Reset|.
   bool IsValid() const;
 
-  void Reset();
-
-  vk::CommandPool GetGraphicsCommandPool() const;
-
+  /// @brief      Creates and returns a new |vk::CommandBuffer|.
+  ///
+  /// An attempt is made to reuse existing buffers (instead of creating new
+  /// ones) by recycling buffers that have been collected by
+  /// |CollectGraphicsCommandBuffer|.
+  ///
+  /// @return     Always returns a new |vk::CommandBuffer|, but if for any
+  ///             reason a valid command buffer could not be created, it will be
+  ///             a `{}` default instance (i.e. while being torn down).
   vk::UniqueCommandBuffer CreateGraphicsCommandBuffer();
 
+  /// @brief      Collects the given |vk::CommandBuffer| for recycling.
+  ///
+  /// The given |vk::CommandBuffer| will be recycled (reused) in the future when
+  /// |CreateGraphicsCommandBuffer| is called.
+  ///
+  /// @param[in]  buffer  The |vk::CommandBuffer| to collect.
+  ///
+  /// @note       This method is a NOP if a different thread created the pool.
```

Comment:
Nit: we use "noop" elsewhere, plus it's more fun to say.

Reply:
Agreed :). Done.